addons/l10n_my_ubl_pint/models/account_edi_xml_pint_my.py

```diff
@@ -0,0 +1,96 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models, _
+
+
+class AccountEdiXmlUBLPINTMY(models.AbstractModel):
+    _inherit = "account.edi.xml.ubl_bis3"
+    _name = 'account.edi.xml.pint_my'
+    _description = "Malaysian implementation of Peppol International (PINT) model for Billing"
+    """
+    * PINT Official documentation: https://docs.peppol.eu/poac/pint/pint/
+    * PINT MY Official documentation: https://docs.peppol.eu/poac/my/pint-my
+    """
+
+    def _export_invoice_filename(self, invoice):
+        # EXTENDS account_edi_ubl_cii
+        return f"{invoice.name.replace('/', '_')}_pint_my.xml"
+
+    def _export_invoice_vals(self, invoice):
+        # EXTENDS account_edi_ubl_cii
+        vals = super()._export_invoice_vals(invoice)
+        vals['vals'].update({
+            # see https://docs.peppol.eu/poac/my/pint-my/bis/#profiles
+            'customization_id': 'urn:peppol:pint:billing-1@my-1',
+            'profile_id': 'urn:peppol:bis:billing',
+        })
+        if invoice.currency_id != invoice.company_id.currency_id:
+            # see https://docs.peppol.eu/poac/my/pint-my/bis/#_tax_in_accounting_currency
+            vals['vals']['tax_currency_code'] = invoice.company_id.currency_id.name  # accounting currency
+        return vals
+
+    def _get_partner_party_tax_scheme_vals_list(self, partner, role):
+        """ [aligned-ibrp-cl-01-my]-Malaysian invoice tax categories MUST be coded using Malaysian codes. """
+        # EXTENDS account_edi_ubl_cii
+        tax_scheme_vals_list = super()._get_partner_party_tax_scheme_vals_list(partner, role)
+        if role == 'supplier':
+            # See https://docs.peppol.eu/poac/my/pint-my/bis/#_seller_tax_identifier
+            tax_scheme_vals_list[0]['company_id'] = partner.sst_registration_number or 'NA'
+            # TIN
+            gst_tax_scheme = tax_scheme_vals_list[0].copy()
+            gst_tax_scheme.update({
+                'company_id': partner.vat,
+                'tax_scheme_vals': {'id': 'GST'},
+            })
+            tax_scheme_vals_list.append(gst_tax_scheme)
+        elif role == 'customer':
+            # See https://docs.peppol.eu/poac/my/pint-my/bis/#_buyer_tax_identifier
+            tax_scheme_vals_list[0]['company_id'] = partner.sst_registration_number or 'NA'
+
+        return tax_scheme_vals_list
+
+    def _get_tax_unece_codes(self, invoice, tax):
+        """ In malaysia, only the following codes can be used: T, E, O
+        https://docs.peppol.eu/poac/my/pint-my/bis/#_tax_category_code
+        """
+        # EXTENDS account_edi_ubl_cii
+        def create_dict(tax_category_code=None, tax_exemption_reason_code=None, tax_exemption_reason=None):
+            return {
+                'tax_category_code': tax_category_code,
+                'tax_exemption_reason_code': tax_exemption_reason_code,
+                'tax_exemption_reason': tax_exemption_reason,
+            }
```

Comment:
I think we can use a defaultdict :D

Reply:
I did it like it was done in the other modules, but sure ^^

Reply:
sorry, I know I was the inventor of this, but it's bad :D